### PR TITLE
TSpline Nodes Documentation - Third set

### PR DESCRIPTION
**Pull request by dariaivanciucova**
### Purpose

A third batch of documentation help files for T-Spline nodes.
The In-Depth description follows [this guidance](https://app.mural.co/t/autodesk2145/m/autodesk2145/1692294995308/331cbe5dd536cfa7530d18679cf94f467f9309eb?sender=63628aa0-4092-42ec-b036-49f61f25d8fa).
This batch includes In-Depth, Example image, and .dyn files for the following _63_ nodes:

**Topology**

- [x] TSplineTopology.DecomposedEdges : _7LMFKLQNCV53W7KLS5QWD3E27NGGA33QPHSXMUGH323WVXWJY3GQ_
- [x] TSplineTopology.DecomposedFaces : _VIA3XNZWZWW3XHWL222NGHWM22VLSA4QXMZCPWZ6JO6G3P7H2WGA_
- [x] TSplineTopology.DecomposedVertices : _GNCPW2XCPOITIUUDNZKISNP347TEGTDWVI5SQTBJX3JWFZTVRXGA_
- [x] TSplineTopology.EdgeByIndex
- [x] TSplineTopology.FaceByIndex
- [x] TSplineTopology.VertexByIndex
- [x] TSplineTopology.BorderEdges
- [x] TSplineTopology.BorderFaces
- [x] TSplineTopology.BorderVertices
- [x] TSplineTopology.EdgesCount
- [x] TSplineTopology.FacesCount
- [x] TSplineTopology.InnerEdges
- [x] TSplineTopology.InnerFaces
- [x] TSplineTopology.InnerVertices
- [x] TSplineTopology.NGonFaces
- [x] TSplineTopology.NonManifoldEdges
- [x] TSplineTopology.NonManifoldVertices
- [x] TSplineTopology.RegularFaces
- [x] TSplineTopology.Regularvertices
- [x] TSplineTopology.StarPointVertices
- [x] TSplineTopology.TPointVertices
- [x] TSplineTopology.VerticesCount

**T-Spline Surface**

- [x] TSplineSurface.CreateMatch(tSplineSurface, tsEdges, curves) : _6ICXLN4V6DNK5KMYTY5LPCJBE27IRW5VOBKCCVFQGO3HST752ZNQ_
- [x] TSplineSurface.CreateMatch(tSplineSurface, tsEdges, brepEdges) : _BUMI5UR5LLKRXP5CUH46L62SN6YIVFGB6FU2PUTKTVDJMTXWXI5Q_
- [x] TSplineSurface.DeserializeFromTSM
- [x] TSplineSurface.DuplicateFaces
- [x] TSplineSurface.EnableSmoothMode
- [x] TSplineSurface.ExportToTSM
- [x] TSplineSurface.ExportToTSS
- [x] TSplineSurface.ExtrudeEdges : _5MRWZW5S4JJMDCAQNQWK32MPFTU5RFV6GLUOOMEN6FLMBO2FOFGQ_
- [x] TSplineSurface.ExtrudeEdgesAlongCurve
- [x] TSplineSurface.ExtrudeFaces
- [x] TSplineSurface.ExtrudeFacesAlongCurve
- [x] TSplineSurface.FillHole
- [x] TSplineSurface.FlattenVertices(vertices) : _WSNXB54TDQEZRTUNRFCTUDK74DHVPQPXBXEYJUZJPW6T2F7XERDQ_
- [x] TSplineSurface.FlattenVertices(vertices, parallelPlane) : _XGSWLBVZ2TGT6X7FZRUWJDGDSBL7JWKDBQYWEJWO7VZOJPJI7OWQ_
- [x] TSplineSurface.FlipNormals
- [x] TSplineSurface.ImportFromTSM(filePath, inSmoothMode) : _SEK72377I2YNUG2LJNOJ6APT3WYI3HJW6FDSVZL3LX2XEMYIA4OQ_
- [x] TSplineSurface.ImportFromTSM(file, inSmoothMode) : _H7WNBOQXEC5RUVELW464ZP2E7CYHSPZ22TJHQJO4AKA4LHFB6N3A_
- [x] TSplineSurface.ImportFromTSS(filePath, inSmoothMode) : _ZBLSFRQNVI3VPC3KTS3XZI6VXSAJAON2ETVE46OLR7NMYOY6INXA_
- [x] TSplineSurface.ImportFromTSS(file, inSmoothMode) : _GD5OJ5UNM2WTBIO7EKGVKBLXL5UMKVOXBJKW2YFKOPVHUIHDCIFA_
- [x] TSplineSurface.Interpolate
- [x] TSplineSurface.MakeUniform
- [x] TSplineSurface.MergeEdges
- [x] TSplineSurface.MoveVertices
- [x] TSplineSurface.PullVertices
- [x] TSplineSurface.Repair
- [x] TSplineSurface.SerializeAsTSM
- [x] TSplineSurface.SlideEdges
- [x] TSplineSurface.Standardize
- [x] TSplineSurface.SubdivideFaces
- [x] TSplineSurface.Thicken(distance, softEdges) : _UHLOMXPCNY3C36FQ45G3HQGKIZLSUE2QX4N7FY7ZCCOEN7F7Q6YA_
- [x] TSplineSurface.Thicken(vector, softEdges) : _USR6ESCX7ACJGZV2YVVIIF7437ZNDU23SUQ6IAHAIPM2YY2FPFGA_
- [x] TSplineSurface.ToBRep
- [x] TSplineSurface.ToMesh
- [x] TSplineSurface.WeldCoincidentVertices
- [x] TSplineSurface.WeldVertices(vertices, newPosition, keepSubdCreases) : _WO4HBR4YYKPQOTXPPFJWGC6B66KH4HDONFO6W7XXS7TACWL7P6YA_
- [x] TSplineSurface.WeldVertices(firstGroup, secondGroup, keepSubdCreases) : _UVMMX2H7AW6KY7XACJHBCRTDYLOGHPZHWREHFNQUMZ6SK33XMHHQ_
- [x] TSplineSurface.IsClosed
- [x] TSplineSurface.IsExtractable
- [x] TSplineSurface.IsInBoxMode
- [x] TSplineSurface.IsStandard
- [x] TSplineSurface.IsWaterTight

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Renamed nodes are submitted with their short names. 

For the `TSplineSurface.SubdivideFaces` the short definition is wrong: 
_Subdivide given faces into four faces each in exact or simple mode depending on given flag value._
For NGon faces the subdivision has a different number of faces (can be 3 or _n_). 

For the `TSplineSurface.CreateMatch(tSplineSurface, tsEdges, curves)` the tooltip for `curves` is wrong: 
_Closed curve loop to create match with_
The curve does not have to be a closed loop.

For `TSplineSurface.CreateMatch(tSplineSurface, tsEdges, brepEdges)` the `continuity` does not currently support G1 & G2 continuity.

### FYIs

@dnenov @Amoursol

### CODE REVIEW

File: doc/distrib/NodeHelpFiles/UVMMX2H7AW6KY7XACJHBCRTDYLOGHPZHWREHFNQUMZ6SK33XMHHQ_img.jpg
Comment (hwahlstrom) [inline doc/distrib/NodeHelpFiles/UVMMX2H7AW6KY7XACJHBCRTDYLOGHPZHWREHFNQUMZ6SK33XMHHQ_img.jpg:1]: Image shows Monocle UI; please remove

File: doc/distrib/NodeHelpFiles/WO4HBR4YYKPQOTXPPFJWGC6B66KH4HDONFO6W7XXS7TACWL7P6YA_img.jpg
Comment (hwahlstrom) [inline doc/distrib/NodeHelpFiles/WO4HBR4YYKPQOTXPPFJWGC6B66KH4HDONFO6W7XXS7TACWL7P6YA_img.jpg:1]: Image shows Monocle UI; please remove

File: doc/distrib/NodeHelpFiles/XGSWLBVZ2TGT6X7FZRUWJDGDSBL7JWKDBQYWEJWO7VZOJPJI7OWQ.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## In Depth
+`TSplineSurface.FlattenVertices(vertices, parallelPlane)` alters the positions of control points for a specified set of vertices by aligning them with a `parallelPlane` provided as input. 
+
+In example below, vertices of a T-Spline plane surface are displaced using `TsplineTopology.VertexByIndex` and `TSplineSurface.MoveVertices` nodes. The surface is then translated to the side for a better preview and used as input for `TSplineSurface.FlattenVertices(vertices, parallelPlane)` node. The result is new surface with selected vertices lying flat on the provided plane.
```
Comment (hwahlstrom) [inline doc/distrib/NodeHelpFiles/XGSWLBVZ2TGT6X7FZRUWJDGDSBL7JWKDBQYWEJWO7VZOJPJI7OWQ.md:4]: "...used as input for **a** `TSplineSurface.FlattenVertices(vertices, parallelPlane)` node."

Comment (hwahlstrom) [inline doc/distrib/NodeHelpFiles/XGSWLBVZ2TGT6X7FZRUWJDGDSBL7JWKDBQYWEJWO7VZOJPJI7OWQ.md:4]: "The result is **a** new surface..."

File: doc/distrib/NodeHelpFiles/ZBLSFRQNVI3VPC3KTS3XZI6VXSAJAON2ETVE46OLR7NMYOY6INXA.md [hunk 2]
```diff
@@ -1,7 +1,5 @@
-<!--- Autodesk.DesignScript.Geometry.TSpline.TSplineSurface.ImportFromTSS(filePath, inSmoothMode) --->
-<!--- ZBLSFRQNVI3VPC3KTS3XZI6VXSAJAON2ETVE46OLR7NMYOY6INXA --->
-## ImportFromTSS(filePath, inSmoothMode) - Documentation
-This documentation file is auto generated by NodeDocumentationMarkdownGenerator, Version=2.14.0.3986, Culture=neutral, PublicKeyToken=null.
-
-For more information about adding documentation to nodes see https://github.com/DynamoDS/Dynamo/wiki/Create-and-Add-Custom-Documentation-to-Nodes
+## In Depth
+In the example below, a spooky T-Spline surface is imported from a T-Spline Scene file using `TSplineSurface.ImportFromTSS` node.
```
Comment (hwahlstrom) [inline doc/distrib/NodeHelpFiles/ZBLSFRQNVI3VPC3KTS3XZI6VXSAJAON2ETVE46OLR7NMYOY6INXA.md:2]: "...using **a** `TSplineSurface.ImportFromTSS` node."

**Review by hwahlstrom — COMMENTED** [review]
LGTM with minor fixes

File: doc/distrib/NodeHelpFiles/GD5OJ5UNM2WTBIO7EKGVKBLXL5UMKVOXBJKW2YFKOPVHUIHDCIFA_img.jpg
[the remark this answers is not in the document]
Comment (dariaivanciucova) [reply doc/distrib/NodeHelpFiles/GD5OJ5UNM2WTBIO7EKGVKBLXL5UMKVOXBJKW2YFKOPVHUIHDCIFA_img.jpg:1]: Confirmed with Aparajit Pratap that the files are ok to be be placed where they currently are, in NodeHelpFiles

**Review by QilongTang — APPROVED** [review]

**Pull request merged**